### PR TITLE
Refactored the `wheel convert` command to not require setuptools

### LINKED ISSUE #510: "wheel convert" depends on setuptools
~~Since 0de0c1a, wheel no longer declares a runtime dependency on setuptools. However, the `convert` command still needs setuptools to work.~~ (Never mind this part, there was never a runtime dependency to start with.)

wheel doesn't declare a runtime dependency on setuptools, but the `convert` command needs it to work. Here's what happens when setuptools is not installed:

```console
> .\venv\Scripts\wheel.exe convert
Traceback (most recent call last):
  File "<frozen runpy>", line 198, in _run_module_as_main
  File "<frozen runpy>", line 88, in _run_code
  File ".\venv\Scripts\wheel.exe\__main__.py", line 7, in <module>
  File ".\venv\Lib\site-packages\wheel\cli\__init__.py", line 91, in main
    args.func(args)
  File ".\venv\Lib\site-packages\wheel\cli\__init__.py", line 29, in convert_f
    from .convert import convert
  File ".\venv\Lib\site-packages\wheel\cli\convert.py", line 10, in <module>
    from ..bdist_wheel import bdist_wheel
  File ".\venv\Lib\site-packages\wheel\bdist_wheel.py", line 24, in <module>
    import pkg_resources
ModuleNotFoundError: No module named 'pkg_resources'
```

### PR DESCRIPTION
**Pull request by agronholm**
This also aims to improve the correctness of the egg and bdist_wininst conversions.

Fixes #510.

### CODE REVIEW

**Comment by agronholm** [comment]
Gotta say, I spent _way_ too much time on this feature that probably no one will use anymore...

File: tests/cli/test_convert.py [hunk 1]
```diff
@@ -1,28 +1,230 @@
...
+import pytest
+from _pytest.fixtures import SubRequest
+from pytest import TempPathFactory
 
+import wheel
+from wheel.cli.convert import convert, egg_filename_re
+from wheel.wheelfile import WHEEL_INFO_RE, WheelFile
 
-def test_egg_re():
+PKG_INFO = dedent(
```
[the remark this answers is not in the document]
Comment (agronholm) [reply tests/cli/test_convert.py:16]: Could you point me to a project that has one (if you know one off the top of your head)?

[the remark this answers is not in the document]
Comment (agronholm) [reply tests/cli/test_convert.py]: Ah well, I added a multi-line License header. All good now?

**Review by henryiii — APPROVED** [review]
I don’t have a use for the command at the moment, but code looks reasonable.

**Comment by agronholm** [comment]
Thanks for the review!

**Pull request merged**